alpha/lib/model/LiveEntry.php

```diff
@@ -480,20 +480,37 @@ public function isCurrentlyLive($currentDcOnly = false)
 		if(!count($liveEntryServerNodes))
 			return false;
 
+		$primaryEsn = null;
+		$secondaryEsn = null;
+
 		/* @var LiveEntryServerNode $liveEntryServerNode*/
 		foreach($liveEntryServerNodes as $liveEntryServerNode)
```

Comment:
Break into smaller function getEsnFromServerNodes(assay liveEntryServerNodes)
When primary+secondary is first found , you can stop the loop.